### PR TITLE
Cypress Setup for Xero app 2

### PR DESCRIPTION
**Pull request by anishfyle**

### CODE REVIEW

File: cypress/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "fyle-xero-app-e2e-test",
+  "version": "0.0.0",
+  "private": true,
+  "dependencies": {
+    "cypress": "^10.3.0"
```
Comment (ashwin1111) [inline cypress/package.json:6]: let's use latest cypress version

File: .github/e2e-test-staging.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            src:
+              - 'src/**'
```
Comment (ashwin1111) [inline .github/e2e-test-staging.yml:30]: include cypress folder as well, any changes in cypress test shouldn't skip the check

File: .github/e2e-test-staging.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+      COMMIT_INFO_MESSAGE: ${{github.event.pull_request.title}}
+      COMMIT_INFO_SHA: ${{github.event.pull_request.head.sha}}
+      APP_URL: ${{ secrets.APP_URL }}
+      API_URL: ${{ secrets.API_URL }}
+      E2E_TESTS_ENV: ${{ secrets.E2E_TESTS_ENV }}
+      E2E_TESTS_REFRESH_TOKEN: ${{ secrets.E2E_TESTS_REFRESH_TOKEN }}
+      E2E_TESTS_ORG_ID: ${{ secrets.E2E_TESTS_ORG_ID }}
```
Comment (ashwin1111) [inline .github/e2e-test-staging.yml:57]: what about workspace secrets?

File: package.json [hunk 4]
```diff
@@ -31,10 +32,11 @@
     "@rxweb/reactive-form-validators": "^2.1.6",
     "@sentry/angular": "^6.19.7",
     "@sentry/tracing": "^6.19.7",
+    "cypress": "^12.0.2",
```
Comment (ashwin1111) [inline package.json:35]: latest

Comment (ashwin1111) [reply package.json:35]: let's add it in dev dependency

Comment (anishfyle) [reply package.json:35]: > let's add it in dev dependency

cypress in both dev dependency and dependancies ?

File: .github/e2e-test-staging.yml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+          message-text: ':pepe-dance: End to End tests (on staging) for Xero just got completed. I have placed the screenshots and videos in this thread :pepe-dance:'
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+      COMMIT_INFO_MESSAGE: ${{github.event.pull_request.title}}
+      COMMIT_INFO_SHA: ${{github.event.pull_request.head.sha}}
+      APP_URL: ${{ secrets.APP_URL }}
+      API_URL: ${{ secrets.API_URL }}
+      E2E_TESTS_WORKSPACE_ID_1: ${{ secrets.E2E_TESTS_WORKSPACE_ID }}
+      E2E_TESTS_WORKSPACE_ID_2: ${{ secrets.E2E_TESTS_WORKSPACE_ID }}
```
Comment (ashwin1111) [inline .github/e2e-test-staging.yml:108]: ${{ secrets.E2E_TESTS_WORKSPACE_ID_2 }}

File: .github/e2e-test-staging.yml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+          channels: 'integration_github_events'
+          message-text: ':pepe-dance: End to End tests (on staging) for Xero just got completed. I have placed the screenshots and videos in this thread :pepe-dance:'
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+      COMMIT_INFO_MESSAGE: ${{github.event.pull_request.title}}
+      COMMIT_INFO_SHA: ${{github.event.pull_request.head.sha}}
+      APP_URL: ${{ secrets.APP_URL }}
+      API_URL: ${{ secrets.API_URL }}
+      E2E_TESTS_WORKSPACE_ID_1: ${{ secrets.E2E_TESTS_WORKSPACE_ID }}
```
Comment (ashwin1111) [inline .github/e2e-test-staging.yml:107]: `${{ secrets.E2E_TESTS_WORKSPACE_ID_1 }}`

File: .github/e2e-test-staging.yml [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+      E2E_TESTS_WORKSPACE_ID_2: ${{ secrets.E2E_TESTS_WORKSPACE_ID }}
+      E2E_TESTS_ENV: ${{ secrets.E2E_TESTS_ENV }}
+      E2E_TESTS_REFRESH_TOKEN_1: ${{ secrets.E2E_TESTS_REFRESH_TOKEN }}
```
Comment (ashwin1111) [inline .github/e2e-test-staging.yml:110]: same, please check all secrets once for this

File: .github/e2e-test-staging.yml [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+          install: true
+          runTests: false
+      env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+      COMMIT_INFO_MESSAGE: ${{github.event.pull_request.title}}
+      COMMIT_INFO_SHA: ${{github.event.pull_request.head.sha}}
+      APP_URL: ${{ secrets.APP_URL }}
+      API_URL: ${{ secrets.API_URL }}
+      E2E_TESTS_ENV: ${{ secrets.E2E_TESTS_ENV }}
```
Comment (ashwin1111) [inline .github/e2e-test-staging.yml:56]: install job doesn't require these secrets

**Review by ashwin1111 — APPROVED** [review]

File: .github/e2e-test-staging.yml (hunk 8 above)
Comment (anishfyle) [reply .github/e2e-test-staging.yml:56]: deleting these 4

**Pull request merged**